### PR TITLE
Remove dsi_data from publishers

### PR DESCRIPTION
**Pull request by cesidio**

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3092.london.cloudapps.digital>

File: app/controllers/publishers/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -114,17 +114,10 @@ def allowed_la_user?
...
+  def use_school_group_if_available(publisher, organisation)
+    return unless organisation.is_a?(School)
 
-    school = School.find_by(urn: school_urn)
-    school_group = school&.school_groups&.first
-    return unless school_group
-
-    session.update(publisher_organisation_id: school_group.id) if
-      publisher_trusts.include?(school_group.uid) || publisher_local_authorities.include?(school_group.local_authority_code)
+    publisher_organisation = publisher.organisations.school_groups.find { |school_group| school_group.schools.include?(organisation) }
```
Comment (cpjmcquillan) [inline app/controllers/publishers/omniauth_callbacks_controller.rb:120]: can we do `publisher.school_groups`? I can't remember 😂

**Review by cpjmcquillan — APPROVED** [review]

File: app/controllers/publishers/omniauth_callbacks_controller.rb (hunk 1 above)
Comment (cesidio) [reply app/controllers/publishers/omniauth_callbacks_controller.rb:120]: nope we cannot

Comment (cpjmcquillan) [reply app/controllers/publishers/omniauth_callbacks_controller.rb:120]: 😞

**Pull request merged**